### PR TITLE
Feature/aced rbac

### PR DESCRIPTION
**Pull request by bwalsh**
This PR:  

* refactors policy names - see https://github.com/ACED-IDP/gen3-helm/blob/develop/docs/aced-RBAC.md
* adds `jobs ls|get|import`
* installed at:  https://pypi.org/project/gen3-util/0.0.7/

### CODE REVIEW

File: docs/project-workflow.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+```text
+# copy the metadata to the bucket and publish the metadata to the portal
+gen3_util meta publish  /tmp/test-myproject  bucket://aced-development-ohsu-data-bucket --project_id test-myproject
+
+```
+
+
+## View in portal
+<img alt="image" src="https://github.com/ACED-IDP/data_model/assets/47808/133ef835-63d6-473e-80ad-9c4e0de62651">
```
Comment (lbeckman314) [inline docs/project-workflow.md:50]: Great docs — very clear.

File: gen3_util/config/config.py [hunk 2]
```diff
@@ -10,7 +11,11 @@
 
 def default():
     """Load config from installed package."""
-    return Config(**yaml.safe_load(pkg_resources.open_text(gen3_util, 'config.yaml').read()))
+    if sys.version_info[:3] <= (3, 9):
+        return Config(**yaml.safe_load(pkg_resources.open_text(gen3_util, 'config.yaml').read()))
```
Comment (lbeckman314) [inline gen3_util/config/config.py:15]: I'm not familiar with the `sys.version_info[:3] <= (3, 9)` and `**yaml` syntax — is this just checking package versions and installing the latest ones?

File: gen3_util/files/uploader.py [hunk 3]
```diff
@@ -42,15 +55,19 @@ def _upload_file_to_signed_url(file_name, md5sum, metadata, signed_url):
...
 
 def _update_indexd(attachment, bucket_name, document_reference, duplicate_check, index_client, md5sum, object_name,
-                   program, project, metadata=None):
+                   program, project, metadata=None, specimen_id=None, patient_id=None, task_id=None, observation_id=None):
     hashes = {'md5': md5sum}
     assert 'id' in document_reference, document_reference
     guid = document_reference['id']
     if metadata is None:
         metadata = {
             **{
```
Comment (lbeckman314) [inline gen3_util/files/uploader.py:64]: Ah, the `**` double stars! These aren't varargs are they?

File: requirements.txt [hunk 4]
```diff
@@ -21,13 +21,14 @@ requests
 # modern python dataclasses
 pydantic
 # lock version
-pydantic-yaml==0.11.2
+pydantic_yaml
+
+# for 3.11 - see https://github.com/yaml/pyyaml/issues/724
+PyYAML==5.3.1
```
Comment (lbeckman314) [inline requirements.txt:27]: Ah, nice catch this issue with PyYAML has been popping up everywhere it seems!

**Review by lbeckman314 — APPROVED** [review]
Really impressive changes, will need to spend some time using this to get acquainted with the new changes but looks really good.

Able to install 0.0.7rc1 (with one non-blocking yaml error included below — might be due to how I'm installing):

```sh
python3 -m venv venv ; source venv/bin/activate

python setup.py install

# running install
# ...
# ext/_yaml.h:2:10: fatal error: 'yaml.h' file not found
#include <yaml.h>
#          ^~~~~~~~
# 1 error generated.
# Error compiling module, falling back to pure Python
...
# Using /Users/beckmanl/code/gen3_util/venv/lib/python3.9/site-packages
# Finished processing dependencies for gen3-util==0.0.7rc1

gen3_util
msg: Version 0.0.7rc1
```

File: docs/project-workflow.md
Comment (lbeckman314) [inline docs/project-workflow.md:1]: Great docs — very clear.

File: gen3_util/config/config.py (hunk 2 above)
Comment (bwalsh) [reply gen3_util/config/config.py:15]: `**`: aka turn a dictionary into keyword arguments
sys.version_info[:3] <= (3, 9) : check python version

File: gen3_util/files/uploader.py (hunk 3 above)
Comment (bwalsh) [reply gen3_util/files/uploader.py:65]: yes, above

File: requirements.txt (hunk 4 above)
Comment (bwalsh) [reply requirements.txt:27]: yes

**Comment by matthewpeterkort** [comment]
Some of the older pytests are failing because the commands have changed since last PR, or some of the tests are expected to fail for one reason or another. 7 tests failed 31 tests passed:

1. `pytest tests/integration/test_files.py::test_files_upload` fails because the command `gen3_util meta  import` no longer exists 
2. `pytest tests/integration/test_access.py::test_files_upload_bad_source` fails because even though it returns `Unable to get a presigned URL for download: 404 Client Error: Not Found for url: https://aced-training.compbio.ohsu.edu/user/data/download/foo` the test exits in a code 0 when it should exit with a 1, but this overall seems like the intended result to me.
3. `pytest tests/integration/test_access.py::test_access_workflow`didn't seem to produce any intended error but hit the  `obj, end = self.scan_once(s, idx)` line and went to the`StopIteration` exception so maybe it had already been run once and was supposed to fail.
4. `pytest tests/integration/test_files.py::test_workflow`  errors because old command `gen3_util meta import`
5. `pytest tests/unit/test_cli.py::test_meta` failed because the first help on the `publish` command is splitting halfway through the sentenceat the `from_` and not at the end of the line. The command contents itself seems to be matching though.
6. `pytest tests/integration/test_meta.py::test_incremental_workflow` seems to be hitting a similar error as `pytest tests/integration/test_access.py::test_access_workflow` and failing at the json decode. This looks like an older test. Maybe just needs to be updated with the newer `gen3_util projects add resource test-myproject` style command

**Review by matthewpeterkort — APPROVED** [review]

File: gen3_util/files/cli.py [hunk 5]
```diff
@@ -19,40 +20,75 @@ def file_group(config):
 
 @file_group.command(name="ls")
 @click.pass_obj
-def files_ls(config: Config):
+@click.option('--object_id', default=None, required=False, show_default=True,
+              help="id of the object in the indexd database")
+@click.option('--project_id', default=None, required=False, show_default=True,
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply gen3_util/files/cli.py:25]: ok.  will improve doc in next PR

File: gen3_util/buckets/__init__.py [hunk 6]
```diff
@@ -23,3 +23,15 @@ def get_buckets(config: Config = None, auth: Gen3Auth = None) -> dict:
...
     assert response.status_code == 200, (response.status_code, response.content)
     return response.json()
+
+
+def get_program_bucket(config: Config, program: str, auth: Gen3Auth = None) -> str:
+    """Get the bucket for a program."""
+    buckets = get_buckets(config=config, auth=auth)
+    bucket_name = None
+    for k, v in buckets['S3_BUCKETS'].items():
+        if program in v['programs']:
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply gen3_util/buckets/__init__.py:33]: ok. will improve doc in next PR

File: docs/project-workflow.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+ResearchSubject.ndjson
+Specimen.ndjson
+Task.ndjson
+```
+
+
+## Publish the project metadata to the portal
+
+```text
+# copy the metadata to the bucket and publish the metadata to the portal
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/project-workflow.md:118]: thanks

**Pull request merged**